js/components/utils.js

```diff
@@ -632,43 +632,96 @@ var Utils = {
-
-    // set lang to default lang in skin config
-    language = localization.defaultLanguage;
-
-    // if no default lang in skin config check browser lang settings
-    if (!language) {
-      if (window.navigator.languages) {
-        // A String, representing the language version of the browser.
-        // Examples of valid language codes are: "en", "en-US", "de", "fr", etc.
-        language = window.navigator.languages[0];
-      } else {
-        // window.navigator.browserLanguage: current operating system language
-        // window.navigator.userLanguage: operating system's natural language setting
-        // window.navigator.language: the preferred language of the user, usually the language of the browser UI
-        language =
-          window.navigator.browserLanguage || window.navigator.userLanguage || window.navigator.language;
-      }
+  getLanguageToUse: function(skinConfig, playerParam) {
+    if (!skinConfig) {
+      return '';
+    }
 
-      // remove lang sub-code
-      var primaryLanguage = language.substr(0, 2);
+    const localization = skinConfig.localization;
 
-      // check available lang file for browser lang
-      for (var i = 0; i < localization.availableLanguageFile.length; i++) {
-        availableLanguages = localization.availableLanguageFile[i];
-        // if lang file available set lang to browser primary lang
-        if (primaryLanguage === availableLanguages.language) {
-          language = primaryLanguage;
-        }
-      }
+    let userBrowserLanguage;
+    let isLanguageCodeInAvailablelLanguageFile = false;
+
+    const isUseUserBrowserLanguage = playerParam && playerParam['useUserBrowserLanguage'];
+
+    // if useUserBrowserLanguage is set to true, use language from user browser settings
+    if (isUseUserBrowserLanguage) {
+      userBrowserLanguage = this.getUserBrowserLanguage();
+      isLanguageCodeInAvailablelLanguageFile = !!localization &&
+        this.isLanguageCodeInAvailablelLanguageFile(
+          localization.availableLanguageFile, userBrowserLanguage
+        );
+    }
+
+    if (!userBrowserLanguage || !localization || !isLanguageCodeInAvailablelLanguageFile) {
+      // if useUserBrowserLanguage is not set to true or
+      // useUserBrowserLanguage is set to true but browser language is not checkable or
+      // there is no browser language in availableLanguageFile
+      return this.getDefaultLanguage(localization);
+    }
+    return userBrowserLanguage;
+  },
+
+  /**
+   *
+   * @param {Object} localization - location configuration object
+   * @param {String} localization.defaultLanguage - value for default language
+   * @returns {String} defaultLanguage - language code from file "skin.json" (
+   * before executing function "onSkinMetaDataFetched") or language code that was set
+   * in the page-level parameters (after executing function "onSkinMetaDataFetched")
+   */
+  getDefaultLanguage: function(localization) {
+    let defaultLanguage = '';
```

Comment:
not necessary, but think of writing in a simpler way
```
const defaultLanguage = !!localization && localization.defaultLanguage;
return defaultLanguage || '';
```

Reply:
@ooyala-amohnatkin If you do not mind I would like to save the code, because I want to check if "localization" is an Object, not only that it is exists. :-)

Reply:
This code also checks if it has `defaultLanguage` key and returns it. Otherwise, it will return empty string

Reply:
yes, it is, it checks the key, but not type of "localization" :-)

Reply:
@ooyala-amohnatkin oh, yes, my fault! Thank you!

I like your way!
\+